cmd/kubeadm/.import-restrictions

```diff
@@ -51,14 +51,74 @@
 			"SelectorRegexp": "k8s[.]io/kubernetes/pkg",
 			"AllowedPrefixes": [
 				"k8s.io/kubernetes/pkg/api/legacyscheme",
-				"k8s.io/kubernetes/pkg/apis/autoscaling",
-				"k8s.io/kubernetes/pkg/apis/core",
 				"k8s.io/kubernetes/pkg/api/service",
+				"k8s.io/kubernetes/pkg/api/testapi",
```

Comment:
is this detection accurate? I only see the following k8s.io/kubernetes/pkg packages referenced from cmd/kubeadm:

```
$ go list -json ./cmd/kubeadm/... | jq -r '.TestImports//[] | .[]' | sort -u | grep k8s.io/kubernetes/pkg
k8s.io/kubernetes/pkg/kubelet/apis/config
k8s.io/kubernetes/pkg/proxy/apis/config
```

Reply:
I don't *think* we care about transitive test dependencies... if we know we need to snip all `k8s.io/kubernetes/pkg` dependencies, then we only care about the things under `./cmd/kubeadm/...` (including *_test.go files) that actually reference `k8s.io/kubernetes/pkg`.

Reply:
i actually don't know where the `testapi` import in particular is coming from.

these are indeed the only direct imports it seems:
```
k8s.io/kubernetes/pkg/kubelet/apis/config
k8s.io/kubernetes/pkg/proxy/apis/config
```

> I don't think we care about transitive test dependencies... 

i guess, one example of wanting to track transitive test dependencies is when a _test.go imports a small third party package to perform operation X, but the third party package is using e.g. a method from the docker-sdk and pulls a lot of code in.

Reply:
it would make sense to require recording package dependencies kubeadm's *_test.go files had (like the two listed above), but I don't want to explicitly allow kubeadm to take direct dependencies on all these additional packages

Reply:
@dims WDYT on this topic?

> but I don't want to explicitly allow kubeadm to take direct dependencies on all these additional packages

that can more of a problem for pkg/controller import restrictions i'd say.
for kubeadm we nowadays watch imports pretty well and have the objective to remove all `k/k/pkg`  imports regardless and move out.

i nice feature for import boss would have been to indicate / allow direct vs indirect in separate lists.

Reply:
> for kubeadm we nowadays watch imports pretty well and have the objective to remove all `k/k/pkg` imports regardless and move out.

the issue is that this PR would make it invisible if kubeadm picked up new direct dependencies on any of these referenced packages

Reply:
but this seems to be a problem without this PR already - any indirect entry in import-restrictions will mask further direct coupling with a problem import.

i personally think that this PR should not be blocked on kubeadm, but since it can be a wider problem perhaps we should reconsider.

in any case, my view is that tracking transitive dependencies for test files is slightly better than not doing so.

Reply:
i guess a compromise would be to only enable support for checking this in import-boss, but having this off by default and not touching the import-restrictions files in this PR.

@dims should this be exposed on the CLI as a flag, but off by default?
> arguments.IncludeTestFiles = true

Reply:
@neolit123 Done!